src/controllers/action-queue/mock.js

```diff
@@ -0,0 +1,349 @@
+      const { walletId } = actionOp
+      const wallet = await account.waitForCurrencyWallet(walletId)
+
+      return mockExecutableAction(account, (): ExecutionOutput => {
+        return {
+          effect: {
+            type: 'tx-confs',
+            txId: mockDelayTimestamp(2000, 5000),
+            walletId: walletId,
+            confirmations: 1
+          },
+          broadcastTxs: mockBroadcastTxs(wallet)
+        }
+      })
+    }
+    case 'loan-deposit': {
+      const { walletId } = actionOp
+      const wallet = await account.waitForCurrencyWallet(walletId)
+
+      return mockExecutableAction(account, (): ExecutionOutput => {
+        return {
+          effect: {
+            type: 'tx-confs',
+            txId: mockDelayTimestamp(3000, 5000),
+            walletId: walletId,
+            confirmations: 1
+          },
+          broadcastTxs: mockBroadcastTxs(wallet)
+        }
+      })
+    }
+    case 'loan-repay': {
+      const { walletId } = actionOp
+      const wallet = await account.waitForCurrencyWallet(walletId)
+
+      return mockExecutableAction(account, (): ExecutionOutput => {
+        return {
+          effect: {
+            type: 'tx-confs',
+            txId: mockDelayTimestamp(2000, 5000),
+            walletId: walletId,
+            confirmations: 1
+          },
+          broadcastTxs: mockBroadcastTxs(wallet)
+        }
+      })
+    }
+    case 'loan-withdraw': {
+      const { walletId } = actionOp
+      const wallet = await account.waitForCurrencyWallet(walletId)
+
+      return mockExecutableAction(account, (): ExecutionOutput => {
+        return {
+          effect: {
+            type: 'tx-confs',
+            txId: mockDelayTimestamp(3000, 5000),
+            walletId: walletId,
+            confirmations: 1
+          },
+          broadcastTxs: mockBroadcastTxs(wallet)
+        }
+      })
+    }
+    case 'swap': {
+      const { toWalletId, toTokenId, fromWalletId: walletId } = actionOp
+      const wallet = await account.waitForCurrencyWallet(walletId)
+
+      return mockExecutableAction(account, (): ExecutionOutput => {
+        return {
+          effect: {
+            type: 'address-balance',
+            address: mockDelayTimestamp(5000, 10000),
+            walletId: toWalletId,
+            tokenId: toTokenId
+          },
+          broadcastTxs: mockBroadcastTxs(wallet)
+        }
+      })
+    }
+
+    case 'broadcast-tx': {
+      throw new Error(`No implementation for action type ${actionOp.type}`)
+    }
+    case 'done': {
+      throw new Error(`No implementation for action type ${actionOp.type}`)
+    }
+    case 'fiat-buy': {
+      throw new Error(`No implementation for action type ${actionOp.type}`)
+    }
+    case 'noop': {
+      throw new Error(`No implementation for action type ${actionOp.type}`)
+    }
+    case 'unixtime': {
+      throw new Error(`No implementation for action type ${actionOp.type}`)
+    }
+
+    default: {
+      // $ExpectError
```

Comment:
Since `exhaustiveCheck` is specifically supposed to create a Flow error if you miss stuff, ignoring it seems weird.

Reply:
`exhaustiveCheck` is meant to throw an error if I didn't miss things. It is a tricky to remember when revisiting this, but for clarity: It's inverted logic hack for Flow; it expects an `empty` type which means that when `default` is reachable, then we haven't exhausted the cases and we expect Flow to give us a type error.